src/modules/activities/profile/partials/ProfileForm.tsx

```diff
@@ -0,0 +1,225 @@
+import { Button, Modal, Spinner } from 'flowbite-react';
+import { useForm } from 'react-hook-form';
+import { joiResolver } from '@hookform/resolvers/joi';
+import { useState, useEffect } from 'react';
+import {
+  HiPencil,
+  HiChevronRight,
+  HiChevronLeft,
+  HiX,
+  HiCheck,
+} from 'react-icons/hi';
+import { toast } from 'react-toastify';
+
+import PersonalInformation from './information/PersonalInformation';
+import AddressInformation from './information/AddressInformation';
+import BankInformation from './information/BankInformation';
+
+import profileSchema from '@/api/schema/profile';
+import {
+  useAppDispatch,
+  useAppSelector,
+} from '@/modules/_partials/hooks/useRedux';
+import { updateProfile } from '@/redux/features/profile/profileSlice';
+
+const ProfileForm = () => {
+  const dispatch = useAppDispatch();
+  const { user, updateError, isUpdating } = useAppSelector(
+    state => state.profile,
+  );
+
+  const profile = user || {};
+
+  const [schema, setSchema] = useState(profileSchema.personal);
+  const {
+    register,
+    reset,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({
+    resolver: joiResolver(schema),
```

Comment:
what do you think about adding
```
reValidateMode: "onChange"
````
on useForm, to revalidate on change (keypress)

Reply:
I think this may cause unnecessary renders and then have significant impact on performance.

Reply:
that's the thing, it won't cause any renders!
by the way, it was a suggestion, you can also keep you version!